k4MarlinWrapper/examples/event_display.py

```diff
@@ -18,58 +18,71 @@
 # limitations under the License.
 #
 
-from Gaudi.Configuration import *
+import os
+import sys
+
+from Gaudi.Configuration import INFO
+from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool, LcioEvent, GeoSvc
+from k4FWCore.parseArgs import parser
+
+parser.add_argument(
+    "--inputFiles",
+    action="extend",
+    nargs="+",
+    metavar=["file1", "file2"],
+    help="One or multiple input files",
+)
+parser.add_argument(
+    "--compactFile", help="Compact detector file to use", type=str, default=""
+)
+
+reco_args = parser.parse_known_args()[0]
 
-from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool
 algList = []
+svcList = []
+
+evtsvc = k4DataSvc("EventDataSvc")
+svcList.append(evtsvc)
+
+if reco_args.compactFile:
+    compact_file = reco_args.compactFile
+else:
+    # add the same default as before
+    # TODO: maybe modernize this
+    compact_file = "CLICPerformance/Visualisation/CLIC_o3_v06_CED/CLIC_o3_v06_CED.xml"
+
+geoSvc = GeoSvc("GeoSvc")
+geoSvc.detectors = [compact_file]
+geoSvc.OutputLevel = INFO
+geoSvc.EnableGeant4Geo = False
+svcList.append(geoSvc)
+
+def create_reader(input_files):
```

Comment:
Since I wrote a very similar thing (at least in purpose) for ILDConfig, I think this could be a useful function to have importable from k4MarlinWrapper. Since we already have some python code for which we do that, this should be fairly easy to add there.

Reply:
Yours is not only very similar this is mostly an exact copy of it ;) 

I agree that we should have it importable, I also want to re-use it for the CLD reconstruction. Should I add it to this PR?

Reply:
No real preference whether it is in this PR or another. I will let you decide.